### ISSUE TITLE
tecnoyfoto.com: detection

### ISSUE THREAD

**Issue by ItsProfessional** [uBlockOrigin/uAssets#20926]
### Prerequisites

- [X] This is **NOT** a [YouTube](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [Facebook](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [Twitch](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I forced an update of my filter lists. (Click the ["Purge all caches"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches) button while holding the 'Shift' key, then click the ["Update now"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now) button.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkpoints.

### URL(s) where the issue occurs.

`https://tecnoyfoto.com/camaras-tapo-a-home-assistant`

### Description

CHP
```adb
tecnoyfoto.com##+js(noeval-if, /chp_?ad/)
```

### Other extensions used

None

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![image](https://github.com/uBlockOrigin/uAssets/assets/63961221/c0b0fe08-28eb-4bae-ac55-88c8562403f7)

</details>

### Configuration

<details>
<summary>Details</summary>

```yaml
uBlock Origin: 1.54.0
Firefox: 120
filterset (summary):
 network: 189838
 cosmetic: 116805
 scriptlet: 26096
 html: 1308
listset (total-discarded, last-updated):
 added:
  easylist-annoyances: 4191-95, 2d.21h.8m
  easylist-chat: 165-0, 2d.21h.8m
  fanboy-cookiemonster: 49037-281, 2d.21h.8m
  ublock-cookies-easylist: 577-0, 2d.21h.8m
  easylist-newsletters: 6660-13, 2d.21h.8m
  easylist-notifications: 2877-4, 2d.21h.8m
  fanboy-social: 16161-7, 2d.21h.8m
  ublock-annoyances: 5852-608, 2d.21h.8m
 default:
  user-filters: 0-0, never
  ublock-filters: 36650-109, 1h.2m
  ublock-badware: 7522-10, 1h.2m
  ublock-privacy: 860-4, 1h.2m
  ublock-unbreak: 2200-2, 1h.2m
  ublock-quick-fixes: 191-2, 1h.2m
  easylist: 75665-731, 3d.22h.19m
  adguard-spyware-url: 1338-94, 3d.22h.19m
  easyprivacy: 32964-824, 3d.22h.19m
  urlhaus-1: 9605-0, 17h.37m
  curben-phishing: 78437-29, 17h.38m
  plowe-0: 3719-1, 3d.22h.19m
  LegitimateURLShortener: 2497-157, 1h.2m
filterset (user): [empty]
trustedset:
 added: [array of 1 redacted]
userSettings:
 advancedUserEnabled: true
hiddenSettings:
 trustedListPrefixes: ublock- user-
supportStats:
 allReadyAfter: 3180 ms
 maxAssetCacheWait: 1988 ms
popupPanel:
 blocked: 16
 network:
  amazon-adsystem.com: 1
  google.com: 2
  googlesyndication.com: 3
  googletagmanager.com: 2
  onesignal.com: 1
  sitelock.com: 1
  translate.googleapis.com: 2
  youtube.com: 4
 extended:
  ##a[href^="https://www.facebook.com/sharer/sharer.php"]
  ##a[href^="https://www.pinterest.com/pin/create/link/"]
  ##.sfsi_widget
  ##.back-to-top:not(body):not(html)
  ##.grecaptcha-badge
  #@#[data-ad-module]
  #@#[data-ad-width]
  #@#[data-adblockkey]
  #@#[data-advadstrackid]
  #@#[data-ad-manager-id]
  #@#.Ad-Container
  #@#.sidebar-ad
  #@#.ad-slot
  #@#.ad-placeholder
  ##+js(set-constant, rwt, noopFunc)
  ##+js(trusted-set-cookie, SOCS, CAESHAgBEhJnd3NfMjAyMzA2MTItMF9S…
```
</details>

**Comment by okiehsch** [comment]
Your filter works fine.
The site uses a generic anti adblock script and we fix that script with
`example##+js(acs, eval, replace)`
that way looking at the code we can tell what caused the issue.

**Comment by okiehsch** [comment]
Soryy, I did somehow not notice your first mention.

I mean that that same script has been used - unchanged - for years and I and others have always fixed it using 
`example##+js(acs, eval, replace)`
If you search the list you will find 100 instances of it.
I did not realize that there was a big block of sites using your `##+js(noeval-if, /chp_?ad/)` until now.

Well, seeing as both filters are now used on literally hundreds of sites we can use both interchangeably after all they both fix the same anti adblock script.

Next time you open an issue and suggest your filter `##+js(noeval-if, /chp_?ad/)` I will use your filter.
